cookbooks/aws-parallelcluster-slurm/libraries/storage.rb

```diff
@@ -0,0 +1,120 @@
+  #
+  # @param [any] change a change provided by a change-set. Example of a change:
+  #                     {
+  #                       "parameter": "SharedStorage",
+  #                       "requestedValue": {
+  #                         "MountDir": "/opt/shared/efs/managed/1",
+  #                         "Name": "shared-efs-managed-1",
+  #                         "StorageType": "Efs",
+  #                         "EfsSettings": {
+  #                           "FileSystemId": "fs-123456789"
+  #                         },
+  #                       "currentValue": "-"
+  #                      }
+  def initialize(change)
+    old_value = change["currentValue"]
+    new_value = change["requestedValue"]
+
+    storage_item = new_value.nil? ? old_value : new_value
+
+    # Storage Action
+    @is_mount = (old_value.nil? and !new_value.nil?)
+    @is_unmount = (!old_value.nil? and new_value.nil?)
+
+    @storage_type = storage_item["StorageType"]
+    @storage_settings = storage_item["#{@storage_type}Settings"] || {}
+
+    # Storage Ownership
+    @is_external = @storage_settings.keys.any? { |k| EXTERNAL_STORAGE_KEYS.include?(k) }
+  end
+
+  # Checks if a given shared storage change within a changeset supports live updates.
+  # With live updates we refer to in-place updates that do not required node replacement.
+  # Currently, a live update is supported only in the following cases:
+  #   1. mount/unmount of external EFS
+  #   1. mount/unmount of external FSx
+  #
+  # @return [Boolean] true if the change supports live updates; false, otherwise.
+  def support_live_updates?
+    @is_external and STORAGE_TYPES_SUPPORTING_LIVE_UPDATES.include?(@storage_type)
+  end
+
+  # Returns the string representation of the object.
+  #
+  # @return [String] the string representation of the object.
+  def to_s
+    attributes = instance_variables.reduce [] do |list, attribute|
+      list.append "#{attribute}=#{instance_variable_get(attribute).inspect}"
+    end
+
+    "{#{attributes.join(', ')}}"
+  end
+end
+
+# Checks if the given changes provided by a change-set support live updates.
+# With live updates we refer to in-place updates that do not required node replacement.
+# The decision on the support is demanded to SharedStorageChangeInfo.
+# If the changes are nil, empty or they do not contain any shared storage change,
+# we assume that a live update is supported.
+# Example of a change:
+#   {
+#     "parameter": "SharedStorage",
+#     "requestedValue": {
+#       "MountDir": "/opt/shared/efs/managed/1",
+#       "Name": "shared-efs-managed-1",
+#       "StorageType": "Efs",
+#       "EfsSettings": {
+#         "FileSystemId": "fs-123456789"
+#       },
+#     "currentValue": "-"
+#    }
+#
+# @param [List[change]] changes the list fo changes provided by a change-set.
+# @return [Boolean] true if the change supports live updates; false, otherwise.
+def storage_change_supports_live_update?(changes)
```

Comment:
note: it's possible to unit tests these library functions, 
see for example: `describe 'aws-parallelcluster-environment:libraries:format_directory' do`

Reply:
Good point, thanks. I assumed it was not. I will include it in the next iteration.